.github/workflows/build.yaml

```diff
@@ -0,0 +1,68 @@
+name: Container Images
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  release:
+    type: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=ref,event=pr
+            type=ref,event=branch,branch=main,latest=true
+            type=ref,event=branch,branch=main
+            type=ref,event=branch,branch!=main
+            type=semver,pattern={{version}},branch=main,latest=false
+            type=ref,event=tag
+
+      - name: Show tags
+        run: |
+          git tag
+          echo ${{ steps.meta.outputs.tags }}
+
+      - name: Extract version from tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Log into GitHub Container Registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: ./Dockerfile
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
+            APE_VERSION=main
```

Comment:
It's `main`? That's weird

I would expect this to use `stable-slim`

Reply:
It will be stable-slim when I fix the ape versioning. I'm not 100% familiar with tagging in github actions, so it's a little trial and error, forgive me